### PR TITLE
LPS-177195 Knowledge Base - Pagination page on the search is ignored

### PR DESCRIPTION
**Pull request by locpham97**
Hi team Lima,
Please help me to review this PR.
Ticket: https://issues.liferay.com/browse/LPS-177195
This is passed review in https://github.com/locpham97/liferay-portal/pull/38
Follow @tototrinh solution:

The issue is caused by the logic here:
https://github.com/liferay/liferay-portal/blob/master/modules/apps/knowledge-base/knowledge-base-web/src/main/java/com/liferay/knowledge/base/web/internal/display/context/KBAdminManagementToolbarDisplayContext.java#L477-L482
https://github.com/liferay/liferay-portal/blob/master/modules/apps/knowledge-base/knowledge-base-service/src/main/java/com/liferay/knowledge/base/service/impl/KBArticleServiceImpl.java#L509-L581

She thought about adjusting the logic in the function but with the existing methods, She would probably have to search all KB articles by keywords, filter the search results to check permissions and then paginate that new list. It could cause performance problems. So she comes up with the idea that we should create new Finder methods that allow doing that.

cc:@tototrinh
Thanks.
Loc

### CODE REVIEW

**Comment by liferay-lima** [comment]
ci:test:sf

**Comment by liferay-lima** [comment]
ci:test:relevant

**Comment by Hanlf** [comment]
The above failures are not related to this PR.

File: modules/apps/knowledge-base/knowledge-base-service/src/main/java/com/liferay/knowledge/base/service/persistence/impl/KBArticleFinderImpl.java [hunk 1]
```diff
@@ -127,6 +234,83 @@ protected String replaceWorkflowStatus(String sql, int[] status) {
...
+
+		if (status == WorkflowConstants.STATUS_APPROVED) {
+			wherePredicate = wherePredicate.and(
+				KBArticleTable.INSTANCE.main.eq(Boolean.TRUE));
+		}
+
+		if (status == WorkflowConstants.STATUS_ANY) {
+			wherePredicate = wherePredicate.and(
+				KBArticleTable.INSTANCE.latest.eq(Boolean.TRUE));
+		}
```
Comment (adolfopa) [inline modules/apps/knowledge-base/knowledge-base-service/src/main/java/com/liferay/knowledge/base/service/persistence/impl/KBArticleFinderImpl.java:305]: This looks wrong. In particular, the conditional that checks for "approved" will never be executed.

**Comment by adolfopa** [comment]
@locpham97 The query logic is very unreadable. Can you try to make the queries a bit more explicity so that they read as actual queries. Right now it is very difficult to figure out what they do.

**Comment by tototrinh** [comment]
ci:test:sf

**Comment by tototrinh** [comment]
ci:test:relevant

**Comment by Hanlf** [comment]
The above Poshi test failure is not related to this PR.

**Review by adolfopa — APPROVED** [review]

**Pull request closed without merging**